src/components/ui/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import { LinearProgress, LinearProgressProps } from "react-native-elements";
+import { Text, View } from "react-native";
+import { useStyles } from "./styles";
+import { colors } from "@app/utils";
+
+interface ProgressBarProps
+  extends Omit<LinearProgressProps, "value" | "color" | "variant"> {
+  id: string;
+  type: "info" | "warning" | "success";
+  maxValue: number;
+  curValue?: number;
+  label?: string;
+  hasProgressText?: boolean; // show progress in label
+  labelType?: "info" | "warning" | "success" | "default";
+}
+
+const colorMap = {
+  info: colors.teal,
+  warning: colors.orange,
+  success: colors.success,
+  default: colors.dark,
+};
+
+/**
+ * Determinant progress bar.
+ *
+ * Usage:
+ *
+ * <ProgressBar
+ *  id="progress-bar"
+ *  type="success"
+ *  label="Pending volunteers"
+ *  hasProgressText
+ *  labelType="success"
+ *  curValue={10}
+ *  maxValue={100}
+ * />
+ */
+const ProgressBar: React.FC<ProgressBarProps> = ({
+  id,
+  type,
+  curValue = 0,
+  maxValue,
+  label,
+  hasProgressText,
+  labelType = "default",
+  ...rest
+}) => {
+  const styles = useStyles({
+    labelColor: colorMap[labelType],
+  });
+  const value = curValue / maxValue;
+  const color = colorMap[type];
+  return (
+    <View style={styles.container}>
+      {(label || hasProgressText) && (
+        <Text style={styles.labelText}>
+          {label || "Progress: "} ({`${curValue} / ${maxValue}`})
```

Comment:
will this give numbers with lots of SFs/decimals? like 0.433321

Reply:
This one will just print it out (e.g. 0/10). The actual value itself will be used by the progressbar since it's expecting something between 0 and 1